src/outline.rs

```diff
@@ -322,6 +323,42 @@ impl<'a> OutlineNodeRef<'a> {
         }
     }
 
+    /// Returns the statement for this node
+    #[must_use]
+    pub fn get_statement(&self) -> StatementRef<'_> {
+        match self {
+            OutlineNodeRef::Chapter { database, node_id } => {
+                let address = database.outline_result().tree[*node_id].stmt_address;
+                database.statement_by_address(address)
+            }
+            OutlineNodeRef::Statement { sref, .. } => *sref,
+        }
+    }
+
+    /// Returns the span for this node.
+    /// For statements, this is the statement span.
+    /// For chapters, this covers the whole chaper, until the next one.
+    #[must_use]
+    pub fn get_span(&self) -> Span {
+        match self {
+            OutlineNodeRef::Chapter { database, .. } => {
+                let stmt = self.get_statement();
+                let start = stmt.span().start;
+                let end = if let Some(next_stmt) = self.next_up() {
```

Comment:
The name `next_up` is confusing, it sounds like up in the file. How about `next_after`? or `after_children`

Reply:
I'll combine all in one: `next_after_children`!